runtime/lua/vim/_system.lua

```diff
@@ -321,7 +324,7 @@ function M.run(cmd, opts, on_exit)
   --- @type vim.SystemState
   local state = {
     done = false,
-    cmd = cmd,
+    cmd = vim.deepcopy(cmd),
```

Comment:
Expose `cmd` on `SystemObj`, because it's a basic info that may be needed by consumers (which didn't necessarily originate the `system()` invocation).

I also wonder if `vim.SystemObj` and `vim.SystemCompleted` should be one interface.

Reply:
Can revert this if needed. LMK

Reply:
> I also wonder if vim.SystemObj and vim.SystemCompleted should be one interface.

I don't think so personally. Each object is mutually exclusive to the other. Merging them together just creates more entropy since each field would become optional, and I generally prefer return types to be as predictable as possible (i.e. avoid returning possible nils).

Reply:
> Expose cmd on SystemObj, because it's a basic info that may be needed by consumers (which didn't necessarily originate the system() invocation). @lewis6991

As long as there's a concrete use case, then it makes sense. What's the use case here?

Reply:
Use-case is that vim.ui.open returns a SystemObj and the caller (`gx` mapping) wants to print info about the command that was run.